j2cl-maven-plugin/src/main/java/com/vertispan/j2cl/mojo/AbstractCacheMojo.java

```diff
@@ -16,6 +16,9 @@ public abstract class AbstractCacheMojo extends AbstractMojo {
     @Parameter(defaultValue = "${project.build.directory}/gwt3BuildCache", required = true, property = "gwt3.cache.dir")
     private File gwt3BuildCacheDir;
 
+    @Parameter(defaultValue = "${project.build.directory}/j2cl-maven-plugin-local-cache", required = true)
```

Comment:
It is a good question, but they serve different purposes. The gwt3buildCache (should be renamed someday) is explicitly able to be shared between projects, and holds build output that could be reusable across modules, projects. On the other hand, the local cache only is meant to be specific to a single module, and reflects the last N (`N=5` above, but you suggest we change this) rebuilds. Those last rebuilds are references to the gwt3BuildCache, so we know what recent rebuilds were appropriate for _this_ module, and in what order they happened.

If the local cache was global, you could work on project A for a day, then project B for a day, then try to update A and it would try to incrementally rebuild from B's last builds. This won't magically protect you from changing git branches, but in theory tasks that are able to be incremental will notice if too much has changed to be incremental and will do a fresh build.